lib/miq_ldap.rb

```diff
@@ -445,7 +445,9 @@ def self.default_ldap_port(scheme = "ldap")
   end
 
   def self.using_ldap?
-    ::Settings.authentication.mode.include?('ldap')
+    ::Settings.authentication.mode.include?('ldap').tap do |should_warn|
+      $audit_log.warn("MiqLdap is a deprecated feature. Please convert to using external authentication.") if should_warn
```

Comment:
We have a deprecation method that should probably be used in this case.  Also I'm not sure $audit_log is really the right place for this even though that's login oriented.

Reply:
See https://github.com/ManageIQ/manageiq/blob/master/lib/vmdb/deprecation.rb.  Though, that avoids logging deprecations in production on purpose and this is trying to log in production on purpose, so I'm not sure.  @gtanzillo @bdunne Opinions?

Reply:
> that avoids logging deprecations in production on purpose

Yeah, maybe create an `ActiveSupport::Deprecation`, but that could be more effort than it's worth.

Reply:
Thank you @Fryguy and @bdunne. adding an `ActiveSupport::Deprecation` does sound like  a lot of work for a limited usage. In this case this functionality is wanted for the next release only. They the code, along with the warning, would be deleted. @gtanzillo Thoughts?

Reply:
I'm thinking this is ok as is. Considering that this is warning a user that a feature is being deprecated vs. a developer that a class, module or method is being deprecated and it's only being done in one place.